static/js/components/SiteContentEditor.tsx

```diff
@@ -55,9 +56,17 @@ export default function SiteContentEditor(
 
   const site = useWebsite()
   const store = useStore()
-  const refreshWebsiteStatus = () =>
+  const refreshWebsiteStatus = () =>{
     store.dispatch(requestAsync(websiteStatusRequest(site.name)))
 
+    // Updating website's has_site_metadata to true
+    store.dispatch(updateEntities({websiteDetails : (prevValue)=>{ 
+      const updatedSite = {...prevValue[site.name], has_site_metadata:true}
+      let newVal = prevValue;
+      newVal[site.name] = updatedSite;
+      return  newVal}}))
+  }
```

Comment:
Is this required?

I thought the following code was enough to fetch new data from the backend.
```js
store.dispatch(requestAsync(websiteStatusRequest(site.name)))
```

Also, since this piece of code sets `has_site_metadata: true` for any form submission, I think saving a page will enable the button even when the metadata is not present.

Reply:
Yes, I didn't go through the working of WebsiteStatus requests in the code. Will update it soon.